Allow using dash and underscore in domains

samaradel opened the pull request:
### Description

- Allow using dash and underscore in domains
- Update models decorators
- Replace characters in workload names
- Update Domains page and manage gateway dialog with new validators

### Changes

[Screencast from 05-03-25 12:14:34.webm](https://github.com/user-attachments/assets/e718cf87-1f13-4f7e-9c99-32eddaa64fc9)

![image](https://github.com/user-attachments/assets/b1d1a675-ea93-4761-ac4f-f402be29f661)


### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3307

### Tested Scenarios

- Add a dash or underscore in the domain
- Adding a dash and an underscore as the first/last character is not acceptable
- Add 2 or more underscores in the same domain name
- Add 2 or more dashes in the same domain name
- Combine 2 underscores and 2 dashes in the same domain name

### Documentation PR

For UI changes, Please provide the Documentation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### To consider

#### Preliminary Checks:

- [x] Preliminary Checks
  - Does it completely address the issue linked?
  - What about edge cases?
  - Does it meet the specified acceptance criteria?
  - Are there any unintended side effects?
  - Does the PR adhere to the team's coding conventions, style guides, and best practices?
  - Does it integrate well with existing features?
  - Does it impact the overall performance of the application?
  - Are there any bottlenecks or slowdowns?
  - Has it been optimized for efficiency?
  - Has it been adequately tested with unit, integration, and end-to-end tests?
  - Are there any known defects or issues?
  - Is the code well-documented?
  - Are changes to documentation reflected in the code?

#### UI Checks:

- [x] UI Checks
  - If a UI design is provided/ does it follow it?
  - Does every button work?
  - Is the data displayed logical? Is it what you expected?
  - Does every validation work?
  - Does this interface feel intuitive?
  - What about slow network? Offline?
  - What if the gridproxy/graphql/chain is failing?
  - What would a first time user have a hard time navigating here?

#### Code Quality Checks:

- [x] Code Quality Checks
  - Code formatted/linted? Are there unused imports? .. etc
  - Is there redundant/repeated code?
  - Are there conditionals that are always true or always false?
  - Can we write this more concisely?
  - Can we reuse this code? If yes, where?
  - Will the changes be easy to maintain and update in the future?
  - Can this code become too complex to understand for other devs?
  - Can this code cause future integration problems?

### Testing Checklist

- [x] Does it Meet the specified acceptance criteria.
- [x] Test if changes can affect any other functionality.
- [ ] Tested with unit, integration, and end-to-end tests.
- [x] Tested the un-happy path and rollback scenarios.
- [ ] Documentation updated to meet the latest changes.
- [ ] Check automated tests working successfully with the changes.
- [ ] Can be covered by automated tests.

### General Checklist

- [x] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstring
- [x] Screenshots/Video attached (needed for UI changes)

Commit:
Allow using dash and underscore in domains
Update models decorators
Replace characters in workload names
Update Domains page and manage gateway dialog with new validators

0oM4R reviewed (commented):
https://github.com/user-attachments/assets/6b25edc8-c780-4511-8db4-5b985ef20758

I can't delete the domain with dash and underscore while the old domains got deleted normally 


also in the details I got the workload name is that expected ?

<img width="899" alt="image" src="https://github.com/user-attachments/assets/8dba3ed3-a425-476d-8530-cb9cd3ba3772" />

samaradel commented:
For deleting domains, I can't check atm as I added a domain from the client and it sound like it's not supported yet to show on the dashboard, so created an issue for this https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3962

For the domain, it's just like what you added except the workload name. The dash is replaced as it's not supported in zos.

Commit:
- Fix delete domains
- Display FQDN instead of workload name

amiraabouhadid reviewed (commented):
can use underscore and dashes in subdomain name now 
![image](https://github.com/user-attachments/assets/029e5a9e-c843-4f9f-aabc-54bb74c163ef)
however the dash doesn't show in the name, only in the domain field 
![image](https://github.com/user-attachments/assets/cfa393a8-a97c-4bd2-bc8d-90eb6d0322b3)
after deleting gateway, contract remains 
![image](https://github.com/user-attachments/assets/6ba10444-78bc-40fe-b56f-c40848c82c32)
I only see this in the logs 
```Start deleting the gateway deployment with name vm10732_vmjixu2n```
even though domain is deleted from list here 
![image](https://github.com/user-attachments/assets/76d4f46b-3e52-4698-a125-0281545cb0aa)

Commit:
- Fix delete domains
- Display domain instead of workload name in the dialog
- Remove name from thr table

Commit:
Remove workload name from domains table

0oM4R commented:
Listing and deleting work fine.
but name is still exist and it is very confusing 
![image](https://github.com/user-attachments/assets/048885b1-2ec5-41eb-89b6-c283b389af62)
and what if we have another domain with the same name but without the dash
![image](https://github.com/user-attachments/assets/def8bc0c-8b4d-4316-ab2b-a8a8fad1481d)

samaradel commented:
I think I should remove it form details (first Image) but I'm not sure about the rest 
@AhmedHanafy725 what do you think?

AhmedHanafy725 commented:
> Listing and deleting work fine. but name is still exist and it is very confusing ![image](https://private-user-images.githubusercontent.com/62248851/421955448-048885b1-2ec5-41eb-89b6-c283b389af62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzNzA2MTQsIm5iZiI6MTc0MjM3MDMxNCwicGF0aCI6Ii82MjI0ODg1MS80MjE5NTU0NDgtMDQ4ODg1YjEtMmVjNS00MWViLTg5YjYtYzI4M2IzODlhZjYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDA3NDUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3MTQzZDQ1ZjY5MDExNDAxYjg5ODI4YjAxZGI0MzRmZmY1NzFmNTc2YjFlNDkwNTYzOTJiNWE1YTM0ZDE4MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ucMkFh2373kusUkqvGilH6kCn4rsPztQ_Mbt-YSAJo4) 

you can remove the name from these details

> and what if we have another domain with the same name but without the dash ![image](https://private-user-images.githubusercontent.com/62248851/421954523-def8bc0c-8b4d-4316-ab2b-a8a8fad1481d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzNzA2MTQsIm5iZiI6MTc0MjM3MDMxNCwicGF0aCI6Ii82MjI0ODg1MS80MjE5NTQ1MjMtZGVmOGJjMGMtOGI0ZC00MzE2LWFiMmItYThhOGZhZDE0ODFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDA3NDUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1M2IzYTQxMzhkZGY2YmYxZDM4ZWZmNmZiZjZhNWJjYjMwNmQyYTMyOGQ1YjU4MzVkN2RiZDc5NjkzMDYyOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2a7Ll2pHIGU4KnxPxE_72Lpqf2aFcuFG2zqPKXUJZLs)

I didn't get what is the issue (the domain is different)

0oM4R commented:
> > and what if we have another domain with the same name but without the dash ![image](https://private-user-images.githubusercontent.com/62248851/421954523-def8bc0c-8b4d-4316-ab2b-a8a8fad1481d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzNzA2MTQsIm5iZiI6MTc0MjM3MDMxNCwicGF0aCI6Ii82MjI0ODg1MS80MjE5NTQ1MjMtZGVmOGJjMGMtOGI0ZC00MzE2LWFiMmItYThhOGZhZDE0ODFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE5VDA3NDUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1M2IzYTQxMzhkZGY2YmYxZDM4ZWZmNmZiZjZhNWJjYjMwNmQyYTMyOGQ1YjU4MzVkN2RiZDc5NjkzMDYyOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2a7Ll2pHIGU4KnxPxE_72Lpqf2aFcuFG2zqPKXUJZLs)
> 
> I didn't get what is the issue (the domain is different)

Confusing if the workload name is visible anywhere

amiraabouhadid reviewed (approved):
was able to create domains with dashes and underscore, validation works and checks if name exists , deletion also works, great job 
![Screenshot 2025-03-23 at 05 40 28](https://github.com/user-attachments/assets/66c09073-afd4-4835-a0c0-ffdc6b8af8a2)

0oM4R commented:
I still can see the workload name after removing the dash 
![image](https://github.com/user-attachments/assets/54047961-74c1-4cef-bb3f-5eec219b3a22)

ramezsaeed reviewed (approved):
Test scenarios:
- add 2 or more underscores in the same domain name.
- add 2 or more dashes in the same domain name
- combine 2 underscores and 2 dashes in the same domain name
- test delete domains with dashes and underscore

Commit:
Remove domain name from details dialog

0oM4R reviewed (changes requested):
![image](https://github.com/user-attachments/assets/77f7b1b6-0e55-406c-b629-99b0b616383f)

still in domains solution

Commit:
Remove domain name input from domain details dialog

0oM4R reviewed (changes requested):
https://github.com/user-attachments/assets/45ca2525-7197-49a9-8c61-696cfa2940a1

I deployed a domain with a custom domain, but I couldn't delete it.
I think this case needs to be handled

ramezsaeed commented:
> Screen.Recording.2025-04-07.at.11.49.56.AM.mov 
> I deployed a domain with a custom domain, but I couldn't delete it. I think this case needs to be handled

I can see the domain name doesn't have dashes or underscore, why it can't be deleted!

Commit:
- Fix domain display when delete
- Fix gateway before deploy using custom domain

samaradel commented:
@0oM4R @ramezsaeed fixed now

Commit:
Fix

Commit:
remove line added by mistake

Commit:
Fix Removing domain deployment

Commit:
Merge branch 'development' into development_validate_domains
